packages/ui/src/components/vuestic-mixins/FormComponent/cFormComponent.ts

```diff
@@ -0,0 +1,170 @@
+    get () {
+      return error.value || internalError.value
+    },
+    set (value) {
+      internalError.value = value
+    },
+  })
+
+  const computedErrorMessages = computed<Array<any>>({
+    get () {
+      return errorMessages.value ? prepareValidations(errorMessages.value) : internalErrorMessages.value
+    },
+    set (errorMessages) {
+      internalErrorMessages.value = errorMessages
+    },
+  })
+
+  function validate () {
+    computedError.value = false
+    computedErrorMessages.value = []
+
+    if (rules.value && rules.value.length > 0) {
+      prepareValidations(rules.value.flat(), modelValue.value)
+        .forEach((validateResult) => {
+          if (typeof validateResult === 'string') {
+            computedErrorMessages.value.push(validateResult)
+            computedError.value = true
+          } else if (validateResult === false) {
+            computedError.value = true
+          }
+        })
+    }
+
+    return !computedError.value
+  }
+
+  function resetValidation (): void {
+    computedErrorMessages.value = []
+    computedError.value = false
+  }
+
+  function hasError (): boolean {
+    return computedError.value
+  }
+
+  const formComponentListeners = {
+    focus (): void {
+      isFocused.value = true
+    },
+    blur (): void {
+      isFocused.value = false
+      validate()
+    },
+  }
+
+  return {
+    validate,
+    resetValidation,
+    hasError,
+    computedError,
+    computedErrorMessages,
+    isFocused,
+    isFormComponent,
+    formComponentListeners,
+  }
+}
+
+/**
+ * Insert this to `@Options`.
+ * This will add `modelValue`, `rules`, `disabled`, `rules`, `readonly`, `success`, `messages`, `error`, `errorMessages`, `errorCount`, `id`, `name` props.
+ * If you want to use `modelValue` in your component you provide prop manually.
+ *
+ * @example
+ ```
+  @Options({ name: 'Example', ...formComponentOptions })
+  export default class ExampleComponent extends Vue.with(props)
+ ```
+ */
+export const formComponentOptions = {
+  props: {
+    rules: {
+      type: Array,
+      default: () => [],
+    },
+    disabled: {
+      type: Boolean,
+      default: false,
+    },
+    readonly: {
+      type: Boolean,
+      default: false,
+    },
+    success: {
+      type: Boolean,
+      default: false,
+    },
+    messages: {
+      type: Array,
+      default: () => [],
+    },
+    error: {
+      type: Boolean,
+      default: false,
+    },
+    errorMessages: {
+      type: [Array, String],
+      default: undefined,
+    },
+    errorCount: {
+      type: Number,
+      default: 1,
+    },
+    id: {
+      type: [String, Number],
+      default: undefined,
+    },
+    name: {
+      type: [String, Number],
+      default: undefined,
+    },
+    modelValue: {
+      type: undefined,
+      default: undefined,
+      validator: () => {
+        throw new Error('ValidateMixin: `modelValue` prop should be defined in component.')
+      },
+    },
+  },
+}
```

Comment:
These types will not trigger TypeScript validation and auto completion.

I believe we should use type casting and PropType for this to get TypeScript support.

Right now creating new composable methods as well so I could update it in my pull request by myself.